sources/osg/Geometry.js

```diff
@@ -325,15 +325,13 @@ utils.createPrototypeNode(
             state.drawGeometry(this);
             var cachedDraw = this._cacheDrawCall[prgID];
 
-            if (this._useVAO) {
-                if (!this._vao[prgID]) {
-                    state.setVertexArrayObject(null);
-                } else {
-                    if (this._vao[prgID].isDirty()) {
-                        // need vertex array recreation
-                        // ie: lost context
-                        cachedDraw = undefined;
-                    }
+            if (!this._vao[prgID]) {
```

Comment:
doesn't it means that on GPU without VAO state will do a bindVertexArray and it will fail ?
 
https://github.com/cedricpinson/osgjs/blob/cd3737da30daeb63471129edea8009644bc7c60e/sources/osg/State.js#L1084

Reply:
ha, perhaps not already call to that later in the same func https://github.com/cedricpinson/osgjs/blob/cd3737da30daeb63471129edea8009644bc7c60e/sources/osg/Geometry.js#L371

so drawGeometry is a drawGeemetryVAO or bindVertexArray(null) doesn't call VAO functions

Reply:
test 

https://github.com/cedricpinson/osgjs/blob/master/sources/osg/Geometry.js#L29

Reply:
@Kuranes no because this line https://github.com/cedricpinson/osgjs/blob/cd3737da30daeb63471129edea8009644bc7c60e/sources/osg/State.js#L1084 will never be reach. stats._currentVAO will be null so it will return at the first line

Reply:
ha, ok.  so it's  implicit  noop. 
(one must not change behaviour of that test in state unaware that it's a noop in nonVAO case.)

Reply:
Yeah maybe we will add a comment